File: LeetCode_Practice/C++_Solutions/LinkedLists/_203_RemoveLinkedListElements.cpp
#include <iostream>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
    public:
        ListNode *removeElements(ListNode *head, int val)
        {
            ListNode *cur = head;
            if (head == nullptr)
                return nullptr;
            while (head != nullptr && head->val == val)
                head = head->next;
            while (cur != nullptr && cur->next != nullptr)
            {
                if (cur->next->val == val)
                    cur->next = cur->next->next;
                else
                    cur = cur->next;
            }
            return head;
        }
};

File: LeetCode_Practice/Java_Solutions/LinkedLists/_203_RemoveLinkedListElements.java
package LeetCode_Practice.Java_Solutions.LinkedLists;

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

public class _203_RemoveLinkedListElements {
    public ListNode removeElements(ListNode head, int val) {
        ListNode cur = head;
        if (head == null)
            return null;
        while (head != null && head.val == val)
            head = head.next;
        while (cur != null && cur.next != null)
        {
            if (cur.next.val == val)
                cur.next = cur.next.next;
            else
                cur = cur.next;
        }
        return head;
    }
}
